robota_core/repository.py

```diff
@@ -483,7 +513,13 @@ def compare(self, point_1: str, point_2: str) -> List[Diff]:
         """
         if not point_1 + point_2 in self._diffs:
             gitlab_diffs = self.project.repository_compare(point_1, point_2)
-            robota_diffs = [Diff(diff, "gitlab") for diff in gitlab_diffs["diffs"]]
+
+            # base url for the diff
+            diff_url = f'{self.project_url}/-/compare/{point_1}...{point_2}'
```

Comment:
The GitLab API returns a `web_url` key/value pair in the JSON from the compare API call.  See the example response at:

https://docs.gitlab.com/ee/api/repositories.html#compare-branches-tags-or-commits

Using this will insulate the code against changes GitLab may make in its API JSON formats (which does happen - there was a big change last summer when we moved to a version of GitLab that added an extra component into almost all URLs, which caused me a lot of headaches where we did not use the provided URL and was easy as pie where we did).

Reply:
The issue with `web_url` is that it links to a page with the raw command output instead of the UI. [Example](https://github.com/robota-suite/robota-core/compare/c6b521a0d54176b4a0c58d6c29f2a89daece4077...2105503a036423e18e0e3474cce2bf851ba5f4f7.diff).

It's all good if this is acceptable. It would save you from a lot of head-scratching in the future.

Reply:
Yeah ... the keys in the GitLab JSON and the GitHub JSON vary quite a bit.  For the GitHub compare commits JSON, you need to use the "html_url" key value pair.